File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/SelectionMode.java
package ch.rmy.android.http_shortcuts.utils;

import android.content.Intent;

import ch.rmy.android.http_shortcuts.plugin.PluginEditActivity;

public enum SelectionMode {

    NORMAL,
    HOME_SCREEN,
    PLUGIN;

    public static SelectionMode determineMode(Intent intent) {
        if (Intent.ACTION_CREATE_SHORTCUT.equals(intent.getAction())) {
            return HOME_SCREEN;
        }
        if (PluginEditActivity.ACTION_SELECT_SHORTCUT_FOR_PLUGIN.equals(intent.getAction())) {
            return PLUGIN;
        }
        return NORMAL;
    }

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/utils/SelectionMode.kt
package ch.rmy.android.http_shortcuts.utils

import android.appwidget.AppWidgetManager
import android.content.Intent
import ch.rmy.android.http_shortcuts.plugin.PluginEditActivity

enum class SelectionMode {

    NORMAL,
    HOME_SCREEN_SHORTCUT_PLACEMENT,
    HOME_SCREEN_WIDGET_PLACEMENT,
    PLUGIN;

    companion object {

        fun determineMode(action: String?) = when (action) {
            Intent.ACTION_CREATE_SHORTCUT -> HOME_SCREEN_SHORTCUT_PLACEMENT
            AppWidgetManager.ACTION_APPWIDGET_CONFIGURE -> HOME_SCREEN_WIDGET_PLACEMENT
            PluginEditActivity.ACTION_SELECT_SHORTCUT_FOR_PLUGIN -> PLUGIN
            else -> NORMAL
        }
    }
}
